### PR TITLE
Modernize function `DepositCharge` ; remove unused code

### PR DESCRIPTION
**Pull request by RemiLehe**
The function `DepositCharge` is used only the electrostatic solver (but I am planning to use it for the initialization of space-charge fields in the electromagnetic solver).

In the current `dev` branch, this function uses obsolete Fortran function (e.g. that do not work on GPU).
Instead, this PR modernizes `DepositCharge`, by calling existing lower-level functions. In particular:

- The actual charge deposition reuses the code that is used in the EM solver (C++ / works on GPU)
- The fine-to-coarse interpolation reuses the existing functions `interpolateFineToCoarse`. These functions were originally member function of the `WarpX` class, but I made them free functions, and declared them in the new file `WarpXComm.H`.

### CODE REVIEW

File: Source/Evolve/WarpXEvolveES.cpp [hunk 1]
```diff
@@ -179,30 +179,6 @@ void WarpX::zeroOutBoundary(amrex::MultiFab& input_data,
     bndry_data.FillBoundary();
 }
```
Comment (RemiLehe) [inline Source/Evolve/WarpXEvolveES.cpp]: This function is not used in the code.

File: Source/FortranInterface/WarpX_f.H [hunk 2]
```diff
@@ -16,23 +16,19 @@
 
 #if (AMREX_SPACEDIM == 3)
```
Comment (RemiLehe) [inline Source/FortranInterface/WarpX_f.H]: I removed these fortran functions.

File: Source/Particles/WarpXParticleContainer.cpp [hunk 3]
```diff
@@ -503,66 +504,54 @@ WarpXParticleContainer::DepositCharge (WarpXParIter& pti, RealVector& wp,
...
+            if (do_field_ionization){
+                ion_lev = pti.GetiAttribs(particle_icomps["ionization_level"]).dataPtr();
+            } else {
+                ion_lev = nullptr;
+            }
 
-        if (!local) rho[lev]->SumBoundary(gm.periodicity());
+            DepositCharge(pti, wp, ion_lev, rho[lev].get(), 0, 0, np, thread_num, lev, lev);
+        }
+#ifdef _OPENMP
```
Comment (ax3l) [inline Source/Particles/WarpXParticleContainer.cpp]: you can just always write the scope `{}` and `#ifdef` out only the `#pragma omp parallel` and thread count to make the code more clear.

Comment (RemiLehe) [reply Source/Particles/WarpXParticleContainer.cpp]: OK, thanks.
I have another PR where I will be touching this function. I'll fix it then.

**Pull request merged**